lib/fluent/capability.rb

```diff
@@ -0,0 +1,71 @@
+#
+# Fluent
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require "fluent/env"
+
+if Fluent.linux?
+  begin
+    require 'capng'
+  rescue LoadError
+  end
+end
+
+module Fluent
+  class Capability
+    def initialize(target = :current_process, pid_or_path = nil)
+      if defined?(CapNG)
```

Comment:
Without `CapNG`, all methods don't do anything. So separate implementation is better.

```
module Fluent
  if defined?(CapNG)
    class Capability
      def apply(select_set)
        @capng.apply(select_set)
      end
    end
  else
    class Capability
      def apply(select_set)
      end
    end
  end
end
```

Reply:
Fixed in https://github.com/fluent/fluentd/pull/3155/commits/6d247cdafd433b091e63c68b0946f4e361a1db30.